pkgs/top-level/all-packages.nix

```diff
@@ -6078,7 +6078,7 @@ with pkgs;
 
   binutilsCross = assert targetPlatform != buildPlatform; lowPrio (
     if targetPlatform.libc == "libSystem" then darwin.cctools_cross
-    else forcedNativePackages.binutils.override {
+    else forcedNativePackages.binutils-raw.override {
```

Comment:
Both are wrong; I fix in my ghc-android branch: you need to feed in the overridden binutils into the darwin wrapper.